dapp/src/components/TransactionListener.js

```diff
@@ -57,6 +57,82 @@ class TransactionListener extends Component {
       })
       this.save(newTransactions)
     }
+
+    const nonMinedTx = this.props.transactions.filter((t) => !t.mined)
+    if (nonMinedTx.length > 0 && !this.state.wsProvider) {
+      this.startWebsocketListener()
+    } else if (nonMinedTx.length === 0 && this.state.wsProvider) {
+      this.cleanupWebSocketProvider()
+    }
+  }
+
+  /* We have a pending transaction so we start listenening for mint / redeem events
+   * and if a transaction with a new hash from the same account and the same nonce arrives
+   * we know user has dropped and replaced a transaction with a higher gas price one.
+   */
+  async startWebsocketListener() {
+    const wsProvider = new ethers.providers.WebSocketProvider(
+      process.env.ETHEREUM_WEBSOCKET_PROVIDER
+    )
+    const vault = ContractStore.currentState.contracts.vault
+
+    const handlePossibleReplacedTransaction = async (eventTransactionHash) => {
+      const eventTx = await wsProvider.getTransaction(eventTransactionHash)
+
+      if (eventTx.from.toUpperCase() === this.props.account.toUpperCase()) {
+        const nonMinedTx = this.props.transactions.filter((t) => !t.mined)
+
+        nonMinedTx
+          .filter((tx) => tx.nonce)
+          .forEach(async (tx) => {
+            // same nonce detected transaction has been dropped and replaced
+            if (tx.nonce === eventTx.nonce) {
+              const otherTransactions = this.props.transactions.filter(
+                (txx) => txx.hash !== tx.hash
+              )
+              // do a copy otherwise pull state won't be happy
+              const newTx = { ...tx }
+
+              // change the necessary fields
+              newTx.hash = eventTx.hash
+              newTx.isError = false
+              newTx.mined = true
+              newTx.observed = false
+              newTx.blockNumber = eventTx.blockNumber
+
+              const newTransactions = [...otherTransactions, newTx]
```

Comment:
Do we want to show that the Tx got replaced somewhere in the DApp?

Reply:
good question. It definitely is relevant info, just not sure if it is important enough.

@micahalcorn what do you think?

Reply:
When the user "speeds up" the transaction or manually submits another one with the same nonce, will it produce a second transaction card in the info feed or will the data from newTx just overwrite the data from nonMinedTx and take its place? I think the simplest solution is just to replace the data in the card and ignore the nonMinedTx. It might get a little confusing if MetaMask shows both and we only show the new one, but I think it's consistent with the user's intent to replace the transaction instead of showing both.

Reply:
yeah that is what happens. The old transaction data gets replaced with the new tx data. 

If there was really a need to show a dropped transaction I would add some info in the expanded transaction space saying that this transaction replaced the other submitted one. And show an etherscan link of the old one. Though we can probably spend engineering resources on a lot of other stuff first.